### PR TITLE
SDA-8382| Fix: skip replicas when passing tuning configs to edit

### PR DESCRIPTION
**Pull request by andreadecorte**
Align behavior of rosa edit machinepool to not ask for replicas if we have another parameter and we are not in interactive method.

Current behaviour
```
rosa edit machinepool -c ad1 workers --tuning-configs tuned-2
? Replicas: 2
I: Updated machine pool 'workers' on hosted cluster 'ad1'
```

Fixed behaviour
```
rosa edit machinepool -c ad1 workers --tuning-configs tuned-2
I: Updated machine pool 'workers' on hosted cluster 'ad1'
```

Align also behaviour for version and autorepair parameters
Fix autorepair interactive behaviour

### CODE REVIEW

File: cmd/edit/machinepool/nodepool.go [hunk 1]
```diff
@@ -24,11 +24,12 @@ func editNodePool(cmd *cobra.Command, nodePoolID string, clusterKey string, clus
 	isAutoscalingSet := cmd.Flags().Changed("enable-autoscaling")
 	isLabelsSet := cmd.Flags().Changed("labels")
 	isTaintsSet := cmd.Flags().Changed("taints")
-	isLabelOrTaintSet := isLabelsSet || isTaintsSet
 	isVersionSet := cmd.Flags().Changed("version")
 	isAutorepairSet := cmd.Flags().Changed("autorepair")
 	isTuningsConfigSet := cmd.Flags().Changed("tuning-configs")
 
+	isAnyAdditionalParameterSet := isLabelsSet || isTaintsSet || isTuningsConfigSet
```
Comment (lucasponce) [inline cmd/edit/machinepool/nodepool.go:31]: Perhaps worth to extend this to version, and autorepair?

Would you mind checking if that works too (it was somehow on my list but I think is a similar code like this) ?

Thanks

Comment (andreadecorte) [reply cmd/edit/machinepool/nodepool.go]: added and validated version + autorepair behaviour

File: cmd/edit/machinepool/nodepool.go [hunk 2]
```diff
@@ -24,14 +24,16 @@ func editNodePool(cmd *cobra.Command, nodePoolID string, clusterKey string, clus
...
 	isAutorepairSet := cmd.Flags().Changed("autorepair")
 	isTuningsConfigSet := cmd.Flags().Changed("tuning-configs")
 
+	// isAnyAdditionalParameterSet is true if at least one parameter not related to replicas and autoscaling is set
+	isAnyAdditionalParameterSet := isLabelsSet || isTaintsSet || isVersionSet || isAutorepairSet || isTuningsConfigSet
+
 	// if no value set enter interactive mode
 	if !(isMinReplicasSet || isMaxReplicasSet || isReplicasSet || isAutoscalingSet || isLabelsSet || isTaintsSet ||
-		isAutorepairSet || isTuningsConfigSet) {
+		isVersionSet || isAutorepairSet || isTuningsConfigSet) {
```
Comment (gdbranco) [inline cmd/edit/machinepool/nodepool.go:36]: Can we substitute this for `isAnyAdditionalParameterSet`?

Comment (andreadecorte) [reply cmd/edit/machinepool/nodepool.go]: good idea, changed that

**Pull request merged**